packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -0,0 +1,204 @@
+  /**
+   * The content to display in the segmented control.
+   */
+  children?: ReactNode,
+  /**
+   * Whether the segmented control is disabled.
+   */
+  isDisabled?: boolean
+}
+export interface ControlItemProps extends Omit<RadioProps, 'children' | 'className' | 'style' | 'onHoverStart' | 'onHoverEnd' | 'onHoverChange'>, StyleProps {
+  /**
+   * The content to display in the control item.
+   */
+  children?: ReactNode
+}
+
+export const SegmentedControlContext = createContext<ContextValue<SegmentedControlProps, DOMRefValue<HTMLDivElement>>>(null);
+
+const segmentedControl = style<{size: string}>({
+  font: 'control',
+  display: 'flex',
+  backgroundColor: 'gray-100',
+  borderRadius: 'lg',
+  width: 'full'
+}, getAllowedOverrides())
+
+const controlItem = style({
+  ...focusRing(),
+  display: 'flex',
+  gap: 'text-to-visual',
+  forcedColorAdjust: 'none',
+  color: {
+    default: 'gray-700',
+    isHovered: 'neutral-subdued',
+    isSelected: 'neutral',
+    isDisabled: 'disabled',
+    forcedColors: {
+      default: 'ButtonText',
+      isDisabled: 'GrayText',
+      isSelected: 'HighlightText'
+    },
+  },
+  backgroundColor: {
+    isSelected: 'gray-25',
+    forcedColors: {
+      isSelected: 'Highlight',
+      isDisabled: {
+        isSelected: 'GrayText'
+      },
+    },
+  },
+  // the padding should be a little less for segmented controls that only contain icons but not sure of a good way to do that
+  paddingX: 'edge-to-text',
+  height: 32,
+  alignItems: 'center',
+  boxSizing: 'border-box',
+  borderStyle: 'solid',
+  borderWidth: 2,
+  borderColor: {
+    default: 'transparent',
+    isSelected: {
+      default: 'gray-900',
+      isDisabled: 'disabled'
+    },
+    forcedColors: {
+      isDisabled: 'GrayText',
+      isSelected: {
+        default: 'Highlight',
+        isDisabled: 'GrayText'
+      }
+    }
+  },
+  borderRadius: 'control',
+  flexBasis: 0,
+  flexGrow: 1,
+  flexShrink: 0,
+  justifyContent: 'center',
+  whiteSpace: 'nowrap',
+  '--iconPrimary': {
+    type: 'fill',
+    value: 'currentColor'
+  }
+}, getAllowedOverrides())
+
+interface SegmentedControlInternalContextProps extends SegmentedControlProps {
+  register?: (value: string) => void;
+}
+
+interface DefaultSelectionTrackProps {
+  defaultValue?: string,
+  value?: string,
+  children?: ReactNode
+}
+
+const SegmentedControlInternalContext = createContext<SegmentedControlInternalContextProps>({});
+
+function SegmentedControl(props: SegmentedControlProps, ref: DOMRef<HTMLDivElement> ) {
+  [props, ref] = useSpectrumContextProps(props, ref, SegmentedControlContext);
+  let {
+    defaultValue,
+    value
+  } = props
+  let domRef = useDOMRef(ref);
+
+  return (
+    <RadioGroup 
+      {...props}
+      ref={domRef}
+      style={props.UNSAFE_style}
+      className={(props.UNSAFE_className || '') + segmentedControl({size: 'M'}, props.styles)}
+      aria-label={props['aria-label'] || 'Segmented Control'}>
```

Comment:
how do we wanna add translations for this?

Reply:
You can add it to the intl files, just add the english string across all of them so it doesn't crash. it'll get picked up in next translation PR after this PR is merged